src/app.js

```diff
@@ -137,20 +138,41 @@ function setupChromeEvents() {
   chrome.idle.onStateChanged.addListener(state => {
     appState.tester_state = state;
     app.pushState();
-    if (state === 'idle') {
+    if (state !== 'active') {
       checkForWorkInterval = DEFAULT_INTERVAL * 10;
+      const TEN_MINUTES = (60 * 10) * 1000;
       shutOffTimer = setTimeout(() => {
-        if (appState.tester_state === 'idle') {
+        if (appState.tester_state !== 'active') {
           appState.isPolling = false;
           app.togglePolling(appState.isPolling);
           app.pushState();
+          const ONE_MINUTE = 60 * 1000;
+          if (appState.lastRefresh > Date.now() - ONE_MINUTE) { // confirm tab recently was refreshed
+            chrome.tabs.update(appState.workTab.id, {url: ABANDONED_URL});
```

Comment:
Not needed, this only triggers if the tabID was refreshed in the last
minute and the user is idle for last 10 minutes.
Tabs don't close themselves, so if tab refreshed by itself during an idle
period we can safely assume the tab is still open.
For the other note I'll add an idle check to ensure user is still idle when
doing the second check.

On Aug 10, 2016 12:33, "Paul Sweeney Jr" <notifications@github.com> wrote:

> In src/app.js
> <https://github.com/rainforestapp/tester-chrome-extension/pull/238#discussion_r74180951>
> :
>
> >            appState.isPolling = false;
> >            app.togglePolling(appState.isPolling);
> >            app.pushState();
> > +          const ONE_MINUTE = 60 * 1000;
> > +          if (appState.lastRefresh > Date.now() - ONE_MINUTE) { // confirm tab recently was refreshed
> > +            chrome.tabs.update(appState.workTab.id, {url: ABANDONED_URL});
>
> error checking before you go ahead w/ update tab.... Rainforest extension
> doesn't really check to see if tabs is closed or whatever. appState.workTab
> doesn't really update itself.
>
> chrome.tabs.get(appState.workTab.id, tab => {
>   if (chrome.runtime.lastError) {
>     appState.workTab = null;
>   } else {
>       // your code here..
>
>   }
> });
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/rainforestapp/tester-chrome-extension/pull/238/files/8d85175ae8dce4177ae7ec12d48500b1edbbc1bb#r74180951>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJs-6Qk4DxZ4ZFDXGN7Stv6gB58Xb5dMks5qeUaJgaJpZM4I6PlG>
> .
>